kin-devplatform-sdk/src/main/java/kin/devplatform/Kin.java

```diff
@@ -45,53 +32,54 @@
 
 public class Kin {
 
-	private static final String KIN_ECOSYSTEM_STORE_PREFIX_KEY = "kinecosystem_store";
-	private static Kin instance;
+	private static KinEcosystemInitiator initiator;
 
-	private final ExecutorsUtil executorsUtil;
-	private EventLogger eventLogger;
-
-	private Kin() {
-		executorsUtil = new ExecutorsUtil();
+	static {
+		ExecutorsUtil executorsUtil = new ExecutorsUtil();
+		initiator = new KinEcosystemInitiator(executorsUtil);
 	}
 
-	private static Kin getInstance() {
-		if (instance == null) {
-			synchronized (Kin.class) {
-				instance = new Kin();
-			}
-		}
-
-		return instance;
+	/**
+	 * Initialize the sdk with all the resources to start, this function isn't doing a network calls at all. In order to
+	 * recover from process restart all the activities in our side should call this method in onCreate.
+	 *
+	 * @param appContext application context.
+	 */
+	public static void initialize(Context appContext, @NonNull KinEnvironment environment) {
```

Comment:
Too much information.

Reply:
reduced :ok_hand:, we have only `start`.